node/src/components/consensus/era_supervisor.rs

```diff
@@ -443,32 +445,72 @@ where
             return effects;
         }
         if block_header.switch_block() {
-            // TODO: Learn the new weights from contract (validator rotation).
-            let validator_stakes = self.era_supervisor.validator_stakes.clone();
-            self.era_supervisor
-                .current_era_mut()
-                .consensus
-                .deactivate_validator();
-            let new_era_id = block_header.era_id().successor();
-            info!(?new_era_id, "Era created");
-            let results = self.era_supervisor.new_era(
-                new_era_id,
-                Timestamp::now(), // TODO: This should be passed in.
-                validator_stakes,
-                block_header.timestamp(),
-                block_header.height() + 1,
-                *block_header.global_state_hash(),
+            // if the block is a switch block, we have to get the validators for the new era and
+            // create it, before we can say we handled the block
+            let request = GetEraValidatorsRequest::new(
+                (*block_header.global_state_hash()).into(),
+                block_header.era_id().successor().0,
+                ProtocolVersion::V1_0_0,
```

Comment:
Shouldn't the contract runtime know the protocol version, based on the block height/era?

Reply:
I'm not sure where the protocol version should come from. From what I've seen, we're hardcoding version 1.0.0 in most places, so I'm just following the pattern here - maybe I should mark it as a TODO?